src/platform/plugins/shared/workflows_management/public/entities/workflows/store/workflow_detail/thunks/update_workflow_thunk.ts

```diff
@@ -45,16 +47,33 @@ export const updateWorkflowThunk = createAsyncThunk<
       queryClient.invalidateQueries({ queryKey: ['workflows'] });
       queryClient.invalidateQueries({ queryKey: ['workflows', id] });
 
+      // If the update affects YAML, update content in the editor immediately
+      // for better UX, then reload from server to sync
+      if (affectsYamlMetadata(workflow)) {
+        const currentYaml = selectYamlString(getState());
+
+        if (currentYaml) {
+          // Update all fields that were changed, preserving formatting (optimistic update)
+          const updatedYaml = updateWorkflowYamlFields(currentYaml, workflow, workflow.enabled);
+          dispatch(setYamlString(updatedYaml));
+        }
+
+        // Also update the workflow object in the store
+        dispatch(updateWorkflow(workflow));
+
+        // Reload the workflow from server to sync
+        await dispatch(loadWorkflowThunk({ id }));
+      } else {
+        dispatch(updateWorkflow(workflow));
+      }
+
```

Comment:
Interesting solution 👏 
It looks good to me. The only thing that I am not sure about is the `dispatch(updateWorkflow(workflow));` at line 62, will the `workflow` still contain the old yaml inside? Is that right? 
When `loadWorkflowThunk` finishes, it will set the correct value, but during that time, if another action happens, it may use a stale yaml value.
Maybe we should do this?
```ts
dispatch(updateWorkflow({ ...workflow, yaml: updatedYaml ));
```